### DISCUSSION TITLE
Helm charts vs yaml deployments

### DISCUSSION THREAD

**Discussion by cveld** [Azure/api-management-self-hosted-gateway#42] [Q&A]
Can you provide guidance when it is suitable to use the Helm charts for deploying the self-hosted gateway and when it is more appropriate to just use a separate yaml file?

We have created a private image based on the self-hosted gateway image with our private root certificate added.
Are the out of the box helm charts provided in this repository still applicable?

**Comment by tomkerkhove** [comment]
The benefit of Helm is that you, as a user, don't have to worry about the resources that the gateway needs - It is a self-contained Kubernetes application that you can just deploy. You could write & maintain the YAML yourself, but the question is why you would want to manage that if you can let somebody else handle it.

In terms of your custom image, you can still use the Helm chart. [In the configuration](https://artifacthub.io/packages/helm/azure-api-management/azure-api-management-gateway#configuration), you can see that it allows you to specify a specific `image.repository` & `image.tag` which could be your registry.

However, if it needs to be authenticated it would need to support image pull secret (https://github.com/Azure/api-management-self-hosted-gateway/issues/23)

**Comment by tomkerkhove** [comment]
@cveld Did this clarify your question? If so, is it OK to close this issue?
